chroma_api/stratagem.py

```diff
@@ -0,0 +1,145 @@
+        )
+        host_ids = pipe(host_ids, set, list)
+        installed_profiles = map(lambda host: host.server_profile_id, ManagedHost.objects.filter(id__in=host_ids))
+        if not all(map(lambda name: name == "stratagem_server", installed_profiles)):
+            return {
+                "code": "stratagem_server_profile_not_installed",
+                "message": "'stratagem_servers' profile must be installed on all MDT servers.",
+            }
+
+        return {}
+
+
+class StratagemConfigurationValidation(RunStratagemValidation):
+    def is_valid(self, bundle, request=None):
+        required_args = [
+            "interval",
+            "report_duration",
+            "report_duration_active",
+            "purge_duration",
+            "purge_duration_active",
+        ]
+
+        difference = set(required_args) - set(bundle.data.keys())
+
+        if not difference:
+            return super(StratagemConfigurationValidation, self).is_valid(bundle, request)
+
+        return {"__all__": "Required fields are missing: {}".format(", ".join(difference))}
+
+
+class StratagemConfigurationResource(ChromaModelResource):
+    filesystem = fields.CharField(attribute="filesystem_id", null=False)
+    interval = fields.IntegerField(attribute="interval", null=False)
+    report_duration = fields.IntegerField(attribute="report_duration", null=False)
+    report_duration_active = fields.BooleanField(attribute="report_duration_active", null=False)
+    purge_duration = fields.IntegerField(attribute="purge_duration", null=False)
+    purge_duration_active = fields.BooleanField(attribute="purge_duration_active", null=False)
+
+    class Meta:
+        resource_name = "stratagem_configuration"
+        queryset = StratagemConfiguration.objects.all()
+        authorization = DjangoAuthorization()
+        authentication = AnonymousAuthentication()
+        allowed_methods = ["get", "post"]
+        validation = StratagemConfigurationValidation()
+
+    @validate
+    def obj_create(self, bundle, **kwargs):
+        return JobSchedulerClient.configure_stratagem(bundle.data)
+
+
+class RunStratagemResource(Resource):
+    filesystem = fields.CharField(attribute="filesystem_id", null=False)
+    report_duration = fields.IntegerField(attribute="report_duration", null=True)
+    purge_duration = fields.IntegerField(attribute="purge_duration", null=True)
+
+    class Meta:
+        list_allowed_methods = ["post"]
+        detail_allowed_methods = []
+        resource_name = "run_stratagem"
+        authorization = DjangoAuthorization()
+        authentication = AnonymousAuthentication()
+        object_class = dict
+        validation = RunStratagemValidation()
+
+    def get_resource_uri(self, bundle=None, url_name=None):
+        return Resource.get_resource_uri(self)
+
+    @validate
+    def obj_create(self, bundle, **kwargs):
+        fs_identifier = str(bundle.data.get("filesystem"))
```

Comment:
str cast not necessary.

Reply:
I've been casting this way since i'm trying to parse to an int. Basically:
```
(Pdb) int("7", 10)
7
(Pdb) int(7, 10)
*** TypeError: int() can't convert non-string with explicit base
(Pdb) int(str(7), 10)
7
```

Reply:
but i've moved the cast so i'll remove it from here.